lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,235 @@
+                            case 40:
+                                that.focusFirstItem(menu);
+                                return false;
+                                break;
+
+                            case 27:
+                                that.closeDropdownMenus();
+                                break;
+                        }
+                    } else {
+
+                        switch(keyCode) {
+                            case 13:
+                            case 38:
+                            case 40:
+                            case 27:
+                        }
+                    }
+                });
+            },
+
+            previousMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length - 1,
+                    index = items.index(focused),
+                    prev = index - 1;
+
+                if (index === 0) {
+                    items.last().focus();
+                } else {
+                    items.eq(prev).focus();
+                }
+            },
+
+            nextMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length -1,
+                    index = items.index(focused),
+                    next = index + 1;
+
+                if (index === items_count) {
+                    items.first().focus();
+                } else {
+                    items.eq(next).focus();
+                }
+            },
+
+            focusFirstItem: function(menu) {
+                menu.find('.dropdown-item:first .action').focus();
+            },
+
+            closeDropdownMenu: function() {
+                var that = _dropdown;
+                    open = that.opts.page.find(that.opts.menu).not(':focus');
+
+                if (open) {
+                    open
+                        .removeClass(that.opts.menu_active)
+                        .addClass(that.opts.menu_inactive);
+
+                    open.parent()
+                        .find(that.opts.button)
+                        .removeClass(that.opts.button_active)
+                        .attr('aria-expanded', 'false');
+                }
+            },
+
+            closeDropdownMenus: function() {
+                var that = _dropdown;
+                    open = that.opts.page.find(that.opts.menu);
+
+                if (open) {
+                    open
+                        .removeClass(that.opts.menu_active)
+                        .addClass(that.opts.menu_inactive);
+
+                    open.parent()
+                        .find(that.opts.button)
+                        .removeClass(that.opts.button_active)
+                        .attr('aria-expanded', 'false');
+                }
+            },
+
+            openDropdownMenu: function(el) {
+                var that = _dropdown,
```

Comment:
It's just not necessary in this context-- you could directly reference the variable _dropdown. Typically "that" is used to store a reference to "this" when there is a inner function defined, and the inner function needs to be able to reference the original "this". 

http://www.crockford.com/javascript/private.html